src/forwarders.js

```diff
@@ -110,12 +111,12 @@ export default function Forwarders(mpInstance, kitBlocker) {
             ) {
                 for (var attrName in userAttributes) {
                     if (userAttributes.hasOwnProperty(attrName)) {
-                        attrHash = mpInstance._Helpers
-                            .generateHash(attrName)
-                            .toString();
-                        valueHash = mpInstance._Helpers
-                            .generateHash(userAttributes[attrName])
-                            .toString();
+                        attrHash = FilterUtilities.hashUserAttributeKey(
+                            attrName
+                        ).toString();
```

Comment:
I wasn't suggesting modifying `generateHash` but rather the return value of `KitFilterHelpers`'s methods.

Reply:
Just to provide a bit of background (and for future devs, bc I think it's a bit confusing either way we do it bc the backend is inconsistent :(). For some reason, the "normal" filtering requires a comparison of numbers, and the "conditional filtering" requires a string comparison.  So the 2 options are:
1. In the KitFilterHelper's methods, have 2 of the ~10 methods return a string, while the 8 others return a number and comment as to why.  In this case the return values are inconsistent, (which is probably okay despite all being labeled `hashFoo`.  
2. Keep as is if we think it's important for all `hash` methods to return the same, and use the toString() where it is being used.

The more I think of it, I think option 1 (your recommendation) is correct.  But I wanted to share that this is a bit nuanced, and backend driven. This way, the helper is doing everything required to make a proper comparison, and we can comment in the code one place why it is this way.

![image](https://user-images.githubusercontent.com/5377436/236559200-aaad4815-d5d0-4bd8-9cca-77e548f8be89.png)

Reply:
I appreciate the explanation and I think it's great to have this context, especially for posterity.

I made my comment initially since I think in some uses of the hash functions, you're converting to a string, so I just assumed that returning a string in those cases would simplify things.

I still think you and I line up on option 1, but option 2 at least grants us some level of "type safety" without converting this file to TypeScript since we know this usage is converting to string. At least with a comment, we can understand why we're doing this conversion.